File: app/src/androidTest/java/com/vgrec/espressoexamples/bases/TestBase.java
package com.vgrec.espressoexamples.bases;

import androidx.test.rule.ActivityTestRule;

import com.vgrec.espressoexamples.activities.MainActivity;

import org.junit.Before;
import org.junit.Rule;

public class TestBase {
    public MainActivity activity;

    @Rule
    public ActivityTestRule<MainActivity> rule = new ActivityTestRule<>(MainActivity.class);

    @Before
    public void setUp() {
        this.activity = rule.getActivity();
    }
}

File: app/src/androidTest/java/com/vgrec/espressoexamples/bases/TestBase.kt
package com.vgrec.espressoexamples.bases

import androidx.test.rule.ActivityTestRule
import com.vgrec.espressoexamples.activities.MainActivity
import org.junit.Before
import org.junit.Rule

open class TestBase {
    var activity: MainActivity? = null

    @Rule
    @JvmField
    var rule = ActivityTestRule(MainActivity::class.java)

    @Before
    fun setUp() {
        activity = rule.activity
    }
}
